### PR TITLE
Full text search on the opportunities Title and Description fields

### PR DESCRIPTION
**Pull request by alifhughes**
- As mentioned in Issue #6 full text search would like to be added to the `Opportunities` model
- This is so we can search and filter them based on 'fuzzy' search criteria from the [front end](https://github.com/FederationOfTech/Coalesce/blob/master/frontend/coalesce/src/pages/OpportunitiesSearchPage.vue#L110)
- This uses [SearchFilter](https://www.django-rest-framework.org/api-guide/filtering/#searchfilter)
- **Notes:** Interested to see if people think it is better to split out the tests, [I realise I have some further setup logic](https://github.com/FederationOfTech/Coalesce/pull/63/files#diff-1f3c91e74ef2f9000ec3248738be94887f9997a4f119687b6f1694215ba8ef39R36-R44) in the test before actually [testing the search functionality](https://github.com/FederationOfTech/Coalesce/pull/63/files#diff-1f3c91e74ef2f9000ec3248738be94887f9997a4f119687b6f1694215ba8ef39R43-R53)

### CODE REVIEW

**Comment by stefanha** [comment]
One `docker-compose exec api ./manage.py test` failure was found. Otherwise it looks good, the test case could be split up but since it's not too big and no other tests cases need to share the setup, the current code seems fine for now.

```
FAIL: coalesce.opportunities.test.test_views:TestOpportunitiesListTestCase.test_post_request_with_query_params_returns_subset
  vi +45 coalesce/opportunities/test/test_views.py  # test_post_request_with_query_params_returns_subset
    eq_(response.status_code, status.HTTP_201_CREATED)
AssertionError: 403 != 201
```

File: coalesce/config/common.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ class Common(Configuration):
         'django.contrib.sessions',
         'django.contrib.messages',
         'django.contrib.staticfiles',
+        'django.contrib.postgres',
```
Comment (stefanha) [inline coalesce/config/common.py:18]: Is this necessary for `search_fields = ['@FIELD']` to work?

**Comment by alifhughes** [comment]
> One `docker-compose exec api ./manage.py test` failure was found. Otherwise it looks good, the test case could be split up but since it's not too big and no other tests cases need to share the setup, the current code seems fine for now.
> 
> ```
> FAIL: coalesce.opportunities.test.test_views:TestOpportunitiesListTestCase.test_post_request_with_query_params_returns_subset
>   vi +45 coalesce/opportunities/test/test_views.py  # test_post_request_with_query_params_returns_subset
>     eq_(response.status_code, status.HTTP_201_CREATED)
> AssertionError: 403 != 201
> ```

Failed test came after I had merged main's changes into my own and there was some new auth code, sorry about that 🤦 Fixed now

File: coalesce/config/common.py (hunk 1 above)
Comment (alifhughes) [reply coalesce/config/common.py:18]: Yep it seems so, when I remove it and re-run the tests I get this exception:
```python
ERROR Internal Server Error: /api/v1/opportunities/itiesListTestCase.test_post_request_with_query_params_returns_subset            ____________
Traceback (most recent call last):                                                                                                             
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner    
  .......
  django.core.exceptions.FieldError: Unsupported lookup 'search' for CharField or join on the field not permitted.
```

Comment (stefanha) [reply coalesce/config/common.py:18]: Got it, thanks for checking!

**Pull request merged**